### PR TITLE
Emit event when headers are imported

### PR DESCRIPTION
**Pull request by claravanstaden**
Emit the following 2 events when beacon and execution headers are imported, respectively:

- `BeaconHeaderImported`
- `ExecutionHeaderImported`

Resolves [SNO-313](https://linear.app/snowfork/issue/SNO-313)

### CODE REVIEW

**Comment by claravanstaden** [comment]
Screenshot 2022-08-26 at 11 23 40

`BeaconHeaderImported` fields:
- block hash
- slot number

`ExecutionHeaderImported ` fields:
- block hash
- block number

File: parachain/pallets/ethereum-beacon-client/src/lib.rs [hunk 1]
```diff
@@ -44,7 +44,11 @@ pub mod pallet {
 	}
 
 	#[pallet::event]
-	pub enum Event<T> {}
+	#[pallet::generate_deposit(pub(super) fn deposit_event)]
+	pub enum Event<T: Config> {
+		BeaconHeaderImported(H256, u64),
```
Comment (vgeddes) [inline parachain/pallets/ethereum-beacon-client/src/lib.rs:49]: More recent versions of Substrate allow you to name fields:
```suggestion
		BeaconHeaderImported { block_hash: H256, block_number: u64 },
```

Comment (claravanstaden) [reply parachain/pallets/ethereum-beacon-client/src/lib.rs]: Much better, thanks
Screenshot 2022-08-26 at 12 50 04
!

**Review by vgeddes — APPROVED** [review]
Looks good!

**Pull request merged**